src/functions/definefunction_test.ts

```diff
@@ -193,3 +193,32 @@ Deno.test("DefineFunction using an OAuth2 property requests a provider key", ()
     OAuth2Function.export().input_parameters.properties,
   );
 });
+
+Deno.test("DefineFunction using an OAuth2 property allows require_end_user_auth", () => {
```

Comment:
@shapirone maybe we need to tackle the TODO in the test prior to this one before we start accepting extensions to the oauth2 properties?

Reply:
I don't think we should hold back this PR, but fixing that TODO would resolve the [no behavioral difference problem](https://github.com/slackapi/deno-slack-sdk/pull/174/files#r1180758629)